core/iwasm/aot/aot_runtime.h

```diff
@@ -109,6 +109,7 @@ typedef struct AOTModuleInstanceExtra {
 #if WASM_ENABLE_MULTI_MODULE != 0
     bh_list sub_module_inst_list_head;
     bh_list *sub_module_inst_list;
+    WASMModuleInstanceCommon **func_module_insts;
```

Comment:
IMU, `sub_module_inst_list` is a list to store all `AOTSubModInstNode`. Why need a new `WASMModuleInstanceCommon[]` to do the same thing?

Reply:
You are right, `sub_module_inst_list` and `func_module_insts` store the same data. But `func_module_insts` is indexed by the function index, which allows us to directly access the module instance without having to go through `sub_module_inst_list` and lookup by name each time. `sub_module_inst_list` is kept because there are still some places that require a complete and non-duplicated sub-module instances list.